CMakeLists.txt

```diff
@@ -1029,6 +1029,13 @@ if (UNIX)
     set(HAVE_FVISIBILITY ON)
   endif (VISIBILITY)
 
+  # Check for -fno-sanitize=null
+  CHECK_C_COMPILER_FLAG("-fno-sanitize=null" HAVE_FNOSANITIZE_NULL)
+
+  if (HAVE_FNOSANITIZE_NULL)
+    set(USE_FNOSANITIZE_NULL ON)
```

Comment:
Where did the unresolved comment here go?  Still pushing to remove this redundant variable.

Generally we want ON or OFF for cmake variables. The #cmakedefines operate with that. We don't normally want to deal with cmake vars that might be undefined.

Other uses of CHECK_C_COMPILER_FLAG do not turn around and make a new variable.

Reply:
Hmmm. I see `HAVE_TYPELIMITS_CONTROL` is actually used directly. Perhaps I'm wrong in understanding the result type of `CHECK_C_COMPILER_FLAG`. Making the change.

Reply:
`HAVE_TYPELIMITS_CONTROL` has a `#cmakedefine` in configure.h which turns it into a C preprocessor define and lets C code use it.  `#cmakedefine` acts based on CMake boolean rules: if `OFF` or `*-NOTFOUND` it is omitted; else it is defined.

Reply:
Now I use just a single symbol everywhere.

> if OFF or *-NOTFOUND it is omitted; else it is defined.

I had some trouble figuring this out. To test the case when the feature is absent, I set the CHECK_C_COMPILER_FLAG arg to a flag I know isn't supported (some random string), and did a `make -j`. But the HAVE_FNOSANITIZE_NULL constant was still getting defined. This caused me to wrongly assume that if the CHECK_C_COMPILER_FLAG result is false, it would still define the macro, but maybe to "false" or something.

I later figured out that, for my CHECK_C_COMPILER_FLAG change to take effect, I need to actually delete the build dir and run the `cmake` and `make -j` commands again! (just `make clean` or `cmake` doesn't do it).